agent/api/container/container.go

```diff
@@ -463,6 +494,10 @@ func (c *Container) GetNextKnownStateProgression() apicontainerstatus.ContainerS
 		return apicontainerstatus.ContainerStopped
 	}
 
+	if c.GetKnownStatus() == apicontainerstatus.ContainerCreated {
```

Comment:
this doesn't seem to be the best solution, when more non linear states are added to the list. 

One way to handle this would be to maintain map of current state to next state information, and return that here. then we do not have to maintain logic specific to different states here.

Reply:
Cool, will refactor this.